spec/main.spec.js

```diff
@@ -800,6 +800,104 @@ describe('Honeybadger', function() {
     });
   });
 
+  describe('window.onunhandledrejection callback', function() {
+    describe('default behavior', function() {
+      beforeEach(function() {
+        Honeybadger.configure({
+          api_key: 'asdf'
+        });
+      });
+
+      it('reports the promise rejection reason when it is a string', function(done) {
+        let reason = 'Something has gone wrong'
+        let promiseRejection = new PromiseRejectionEvent('unhandledrejection', {
+          promise: Promise.reject(reason),
+          reason
+        })
+        window.onunhandledrejection(promiseRejection);
+
+        afterNotify(done, function() {
+          expect(requests.length).toEqual(1);
+          expect(request.payload.error.class).toEqual('window.onunhandledrejection')
+          expect(request.payload.error.message).toEqual('UnhandledPromiseRejectionWarning: Something has gone wrong');
+          expect(request.payload.error.backtrace).toEqual('UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().')
+        });
+      });
+
+      it('reports the promise rejection reason when it is an Error object', function(done) {
+        let reason = new Error('Something has gone wrong')
+        reason.stack = "the stack trace"
+        let promiseRejection = new PromiseRejectionEvent('unhandledrejection', {
+          promise: Promise.reject(reason),
+          reason
+        })
+        window.onunhandledrejection(promiseRejection);
+
+        afterNotify(done, function() {
+          expect(requests.length).toEqual(1);
+          expect(request.payload.error.class).toEqual('Error')
+          expect(request.payload.error.message).toEqual('UnhandledPromiseRejectionWarning: Error: Something has gone wrong');
+          expect(request.payload.error.backtrace).toEqual('the stack trace\nUnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().')
```

Comment:
Question about this stack trace--I don't think I've seen traces with any kind of additional message/warning at the end; usually it's at the beginning:

```
Error message here
  at url:1:1
  at ...
```

Am I missing something there?

Reply:
I got this message and pattern directly from node. Node also appends this message at the bottom of the stack trace. I'm fine removing it, but it seems like valuable context for errors of this sort. It doesn't seem like browsers have a message like this built in already.

What do you think?

Here is an example:

```
$ node
> new Promise(resolve => { foo() })
Promise {
  <rejected> ReferenceError: foo is not defined
      at resolve (repl:1:26)
      at new Promise (<anonymous>)
      at repl:1:1
      at Script.runInThisContext (vm.js:124:20)
      at REPLServer.defaultEval (repl.js:325:29)
      at bound (domain.js:425:14)
      at REPLServer.runBound [as eval] (domain.js:438:12)
      at REPLServer.onLine (repl.js:650:10)
      at REPLServer.emit (events.js:198:15)
      at REPLServer.EventEmitter.emit (domain.js:481:20) }
> (node:11338) UnhandledPromiseRejectionWarning: ReferenceError: foo is not defined
    at resolve (repl:1:26)
    at new Promise (<anonymous>)
    at repl:1:1
    at Script.runInThisContext (vm.js:124:20)
    at REPLServer.defaultEval (repl.js:325:29)
    at bound (domain.js:425:14)
    at REPLServer.runBound [as eval] (domain.js:438:12)
    at REPLServer.onLine (repl.js:650:10)
    at REPLServer.emit (events.js:198:15)
    at REPLServer.EventEmitter.emit (domain.js:481:20)
(node:11338) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:11338) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

Reply:
Gotcha, thanks for the explanation. I'm thinking that we should leave it out of the browser version. I like the extra context, but our backend doesn't support extra logging at the end of the stack trace, unfortunately.

Reply:
Gotcha, will remove.